mm2src/coins/eth.rs

```diff
@@ -2328,10 +2328,37 @@ impl MarketCoinOps for EthCoin {
         Some(keccak256(&stream.out()).take())
     }
 
-    fn sign_message(&self, message: &str) -> SignatureResult<String> {
+    fn sign_message(&self, message: &str, derivation_path: Option<DerivationPath>) -> SignatureResult<String> {
         let message_hash = self.sign_message_hash(message).ok_or(SignatureError::PrefixNotFound)?;
-        let privkey = &self.priv_key_policy.activated_key_or_err()?.secret();
-        let signature = sign(privkey, &H256::from(message_hash))?;
+
+        let signature = match derivation_path {
+            Some(derivation_path) => {
+                let expected_coin_type = self.priv_key_policy.path_to_coin_or_err()?.coin_type();
```

Comment:
could we check the whole path to coin instead of just the coin type?

Reply:
oh that should be the purpose and the coin_type

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2432/commits/187284d0b1e07ef7d9c243cb35dba210063c2b4d

Reply:
thanks

what about this tho:
```rust
                let expected_path = self.priv_key_policy.path_to_coin_or_err()?;
                let rpc_path = HDPathToCoin::try_from(derivation_path.clone())?;
                if expected_path != &rpc_path {
                    let error = format!(
                        "Derivation path '{:?}' must be equal to '{:?}'",
                        derivation_path, expected_path
                    );
                    return MmError::err(SignatureError::InvalidRequest(error));
                };
```

Reply:
but looks like the tail applies a strict length check: https://github.com/KomodoPlatform/komodo-defi-framework/blob/187284d0b1e07ef7d9c243cb35dba210063c2b4d/mm2src/crypto/src/bip32_child.rs#L233

we can trim the derivation path before `HDPathToCoin::try_from` or use `StandardHDPath` but add a method to it that gets the `HDPathToCoin` from `self`.

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2432/commits/b10fd578c4ff20698123bbad810fdf04ce598fee